app/uk/gov/hmrc/tai/viewModels/TaxCodeViewModelPreviousYears.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.viewModels
+
+import org.joda.time.LocalDate
+import play.api.i18n.Messages
+import uk.gov.hmrc.play.language.LanguageUtils.Dates
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.domain.TaxCodeRecord
+import uk.gov.hmrc.tai.util.ViewModelHelper
+
+
+case class TaxCodeViewModelPreviousYears(title: String,
```

Comment:
I would like to do more to remove some of the duplication. They do however have some different logic and were initially separated because of that.

I agree they could be resolved into one class model in the end